core/controller/src/main/scala/whisk/core/loadBalancer/LoadBalancerService.scala

```diff
@@ -152,43 +140,38 @@ class LoadBalancerService(
      * Creates an activation entry and insert into various maps.
      */
     private def setupActivation(action: ExecutableWhiskAction, activationId: ActivationId, namespaceId: UUID, invokerName: String, transid: TransactionId): ActivationEntry = {
-        // either create a new promise or reuse a previous one for this activation if it exists
+
         val timeout = action.limits.timeout.duration + activeAckTimeoutGrace
```

Comment:
@vvraskin there is a theoretical possibility here in that `ActivationEntry` is more than just the ActivationId. In a concurrent case we could add duplicates.